IKEA_Samla_Inserts.scad

```diff
@@ -343,17 +342,24 @@ module Create_Samla_Insert(width, depth, height, scale_width, scale_depth, width
     }
 }
 
-if (Active_Layer <= Layers) {
+start_layer = Active_Layer == 0 ? 1 : Active_Layer;
+end_layer = Active_Layer == 0 ? Layers : Active_Layer;
+
+for ( layer = [ start_layer : end_layer ] )
+{
     if (Box_Size == "5")
     {
-        Create_Samla_Insert(5_width, 5_depth, 5_height, 5_scale_width, 5_scale_depth, 5_width_handle, 5_depth_handle, 5_width_cutout, 5_depth_cutout, 5_scale_handle, 5_scale_cutout, 5_handle_cutout_height, 5_diameter, 5_diameter2, Resolution);
+        translate( [ 0, layer * 5_depth * 5_scale_depth,0] )
+            Create_Samla_Insert(layer, 5_width, 5_depth, 5_height, 5_scale_width, 5_scale_depth, 5_width_handle, 5_depth_handle, 5_width_cutout, 5_depth_cutout, 5_scale_handle, 5_scale_cutout, 5_handle_cutout_height, 5_diameter, 5_diameter2, Resolution);
     }
     else if (Box_Size == "11")
     {
-        Create_Samla_Insert(11_width, 11_depth, 11_height, 11_scale_width, 11_scale_depth, 11_width_handle, 11_depth_handle, 11_width_cutout, 11_depth_cutout, 11_scale_handle, 11_scale_cutout, 11_handle_cutout_height, 11_diameter, 11_diameter2, Resolution);
+        translate( [ 0, layer * 11_depth * 11_scale_depth,0] )
+            Create_Samla_Insert(layer, 11_width, 11_depth, 11_height, 11_scale_width, 11_scale_depth, 11_width_handle, 11_depth_handle, 11_width_cutout, 11_depth_cutout, 11_scale_handle, 11_scale_cutout, 11_handle_cutout_height, 11_diameter, 11_diameter2, Resolution);
     }
     else if (Box_Size == "22")
     {
-        Create_Samla_Insert(22_width, 22_depth, 22_height, 22_scale_width, 22_scale_depth, 22_width_handle, 22_depth_handle, 22_width_cutout, 22_depth_cutout, 22_scale_handle, 22_scale_cutout, 22_handle_cutout_height, 22_diameter, 22_diameter2, Resolution);
+        translate( [ 0, layer * 22_depth * 22_scale_depth, 0,0] )
+            Create_Samla_Insert(layer, 22_width, 22_depth, 22_height, 22_scale_width, 22_scale_depth, 22_width_handle, 22_depth_handle, 22_width_cutout, 22_depth_cutout, 22_scale_handle, 22_scale_cutout, 22_handle_cutout_height, 22_diameter, 22_diameter2, Resolution);
```

Comment:
I'd rather like to see here "(layer-1)" to center the first layer at the origin as it was before...